src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -104,124 +104,68 @@ int ACA_Dataplane_OVS::update_port_state_workitem(const PortState current_PortSt
-      virtual_mac_address = current_PortConfiguration.mac_address();
-      if (!aca_validate_mac_address(virtual_mac_address.c_str())) {
-        throw std::invalid_argument("virtual_mac_address is invalid");
-      }
-
-      // TODO: cache the subnet information to a dictionary to provide
-      // a faster look up for the next run, only use the below loop for
-      // cache miss.
-      // Look up the subnet configuration to query for tunnel_id
-      for (int j = 0; j < parsed_struct.subnet_states_size(); j++) {
-        SubnetConfiguration current_SubnetConfiguration =
-                parsed_struct.subnet_states(j).configuration();
-
-        ACA_LOG_DEBUG("current_SubnetConfiguration subnet ID: %s.\n",
-                      current_SubnetConfiguration.id().c_str());
-
-        if (parsed_struct.subnet_states(j).operation_type() == OperationType::INFO) {
-          if (current_SubnetConfiguration.id() ==
-              current_PortConfiguration.fixed_ips(0).subnet_id()) {
-            found_network_type = current_SubnetConfiguration.network_type();
-            found_tunnel_id = current_SubnetConfiguration.tunnel_id();
-            if (!aca_validate_tunnel_id(found_tunnel_id, found_network_type)) {
-              throw std::invalid_argument("found_tunnel_id is invalid");
-            }
-
-            found_cidr = current_SubnetConfiguration.cidr();
-            slash_pos = found_cidr.find('/');
-            if (slash_pos == string::npos) {
-              throw std::invalid_argument("'/' not found in cidr");
-            }
-
-            // substr can throw out_of_range and bad_alloc exceptions
-            found_prefix_len = found_cidr.substr(slash_pos + 1);
+    if (!aca_validate_fixed_ips_size(current_PortConfiguration.fixed_ips_size())) {
+      throw std::invalid_argument("PortConfiguration.fixed_ips_size is less than zero");
+    }
 
-            subnet_info_found = true;
-            break;
+    // TODO: cache the subnet information to a dictionary to provide
+    // a faster look up for the next run, only use the below loop for
+    // cache miss.
+    // Look up the subnet configuration to query for tunnel_id
+    for (int j = 0; j < parsed_struct.subnet_states_size(); j++) {
+      SubnetConfiguration current_SubnetConfiguration =
+              parsed_struct.subnet_states(j).configuration();
+
+      ACA_LOG_DEBUG("current_SubnetConfiguration subnet ID: %s.\n",
+                    current_SubnetConfiguration.id().c_str());
+
+      if (parsed_struct.subnet_states(j).operation_type() == OperationType::INFO) {
+        if (current_SubnetConfiguration.id() ==
+            current_PortConfiguration.fixed_ips(0).subnet_id()) {
+          found_network_type = current_SubnetConfiguration.network_type();
+          found_tunnel_id = current_SubnetConfiguration.tunnel_id();
+          if (!aca_validate_tunnel_id(found_tunnel_id, found_network_type)) {
+            throw std::invalid_argument("found_tunnel_id is invalid");
           }
```

Comment:
This block of code naturally fits into a standalone method.

Reply:
Let us try to limit the length of every method (30 is a magic number to me :-)

Reply:
I put this subnet lookup into a method.

As for limit the length of every method, we have some way to go. Will do it one step at a time :)

Reply:
Sounds good.